receiver/splunksearchapireceiver/receiver.go

```diff
@@ -0,0 +1,317 @@
+	id component.ID,
+) *splunksearchapireceiver {
+	return &splunksearchapireceiver{
+		logger:           logger,
+		config:           config,
+		settings:         settings,
+		storageClient:    storage.NewNopClient(),
+		id:               id,
+		checkpointRecord: &EventRecord{},
+	}
+}
+
+func (ssapir *splunksearchapireceiver) Start(ctx context.Context, host component.Host) error {
+	var err error
+	ssapir.client, err = newDefaultSplunkSearchAPIClient(ctx, ssapir.settings, *ssapir.config, host)
+	if err != nil {
+		return err
+	}
+
+	// set cancel function
+	cancelCtx, cancel := context.WithCancel(ctx)
+	ssapir.cancel = cancel
+
+	// create storage client
+	storageClient, err := adapter.GetStorageClient(ctx, host, ssapir.config.StorageID, ssapir.id)
+	if err != nil {
+		return fmt.Errorf("failed to get storage client: %w", err)
+	}
+	ssapir.storageClient = storageClient
+
+	err = ssapir.initCheckpoint(cancelCtx)
+	if err != nil {
+		return fmt.Errorf("failed to initialize checkpoint: %w", err)
+	}
+	go ssapir.runQueries(cancelCtx)
+	return nil
+}
+
+func (ssapir *splunksearchapireceiver) Shutdown(ctx context.Context) error {
+	ssapir.logger.Debug("shutting down logs receiver")
+	if ssapir.cancel != nil {
+		ssapir.cancel()
+	}
+
+	if ssapir.storageClient != nil {
+		if err := ssapir.checkpoint(ctx); err != nil {
+			ssapir.logger.Error("failed checkpoint", zap.Error(err))
+		}
+		return ssapir.storageClient.Close(ctx)
+	}
+	return nil
+}
+
+func (ssapir *splunksearchapireceiver) runQueries(ctx context.Context) {
+	for _, search := range ssapir.config.Searches {
+		// set current search query
+		ssapir.checkpointRecord.Search = search.Query
+
+		// set default event batch size (matches Splunk API default)
+		if search.EventBatchSize == 0 {
+			search.EventBatchSize = splunkDefaultEventBatchSize
+		}
+
+		// create search in Splunk
+		searchID, err := ssapir.createSplunkSearch(search)
+		if err != nil {
+			ssapir.logger.Error("error creating search", zap.Error(err))
+			return
+		}
+
+		// wait for search to complete
+		if err = ssapir.pollSearchCompletion(ctx, searchID); err != nil {
+			ssapir.logger.Error("error polling for search completion", zap.Error(err))
+			return
+		}
+
+		for {
+			ssapir.logger.Info("fetching search results")
+			results, err := ssapir.getSplunkSearchResults(searchID, offset, search.EventBatchSize)
+			if err != nil {
+				ssapir.logger.Error("error fetching search results", zap.Error(err))
+			}
+			ssapir.logger.Info("search results fetched", zap.Int("num_results", len(results.Results)))
+
+			// parse time strings to time.Time
+			earliestTime, err := time.Parse(time.RFC3339, search.EarliestTime)
+			if err != nil {
+				ssapir.logger.Error("earliest_time failed to be parsed as RFC3339", zap.Error(err))
+			}
+
+			latestTime, err := time.Parse(time.RFC3339, search.LatestTime)
+			if err != nil {
+				ssapir.logger.Error("latest_time failed to be parsed as RFC3339", zap.Error(err))
```

Comment:
I would pull these up one level out of this for loop since these are always the same for this specific `search`. I'd also consider not worrying about this error since we already verify it parses to this format in the config validation, so this could be simplified to: 
```suggestion
			// parse time strings to time.Time
			// not checking error; previously verified in config validation
			earliestTime, _ := time.Parse(time.RFC3339, search.EarliestTime)
			latestTime, _ := time.Parse(time.RFC3339, search.LatestTime)
```

Or you could update the `Searches` struct to contain a couple non exported, non tagged time.Time objects, like `parsedEarliestTime` and you set those values during config validation and reference them instead.